apis/bigqueryconnection/v1alpha1/connection_reference.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	refsv1beta1 "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ refsv1beta1.Resolver = &BigQueryConnectionConnectionRef{}
+
+/*
+type ErrNoServerID struct{}
+
+func (e *ErrNoServerID) Error() string {
+	return "BigQueryConnectionConnection service-generated ID is not assigned yet"
+}*/
+
+type BigQueryConnectionConnectionRef struct {
+	// A reference to an externally managed BigQueryConnectionConnection resource.
+	// Should be in the format `projects/<projectID>/locations/<location>/connections/<connectionID>`.
+	External string `json:"external,omitempty"`
+
+	// The `name` of a `BigQueryConnectionConnection` resource.
+	Name string `json:"name,omitempty"`
+	// The `namespace` of a `BigQueryConnectionConnection` resource.
+	Namespace string `json:"namespace,omitempty"`
+}
+
+func (r *BigQueryConnectionConnectionRef) ValidateExternal() error {
+	if r.External == "" {
```

Comment:
Nit: a method called Validate probably should not mutate the value

Reply:
fixed. Good catch. I got lost on changing the code name and input/output back and forth.